src/Compilers/CSharp/Portable/Symbols/SubstitutedNamedTypeSymbol.cs

```diff
@@ -159,6 +159,10 @@ internal override ImmutableArray<NamedTypeSymbol> GetInterfacesToEmit()
             throw ExceptionUtilities.Unreachable;
         }
 
+        internal override bool GetUnificationUseSiteDiagnosticRecursive(ref DiagnosticInfo result, Symbol owner, ref HashSet<TypeSymbol> checkedTypes)
+            => OriginalDefinition.GetUnificationUseSiteDiagnosticRecursive(ref result, owner, ref checkedTypes)
+               || ContainingType.GetUnificationUseSiteDiagnosticRecursive(ref result, owner, ref checkedTypes);
```

Comment:
>ContainingType [](http://example.com/codeflow?start=18&length=14)

it looks like I was wrong and we shouldn't recurse into the containing type here given that base implementation doesn't do that. There is also an explicit comment about that there:
```
            // When we are binding a name with a nested type, Goo.Bar, then we ask for
            // use-site errors to be reported on both Goo and Goo.Bar. Therefore we should
            // not recurse into the containing type here; doing so will result in errors
            // being reported twice if Goo is bad.
```